string command
string intermediate
---
bool ack
